### PR TITLE
feat: twilio provider

### LINKED ISSUE #446: Add Twillo provider
Twillo Provider should be able to send SMS'es.

Things to notice:
1. Required scopes to send SMS and `validate_scopes` to validate them
2. Docs

You can see `jira_provider` for example on validate_scopes

### PR DESCRIPTION
**Pull request by sc231997**
Closes #446 
/claim #446

## 📑 Description
Twilio Provider added
Documentation added

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by shahargl** [comment]
hey @sc231997, nice work! Can you please add the scope validation? 

You can look at `jira_provider`/`grafana_provider` for example.
You basically need two things:
1. Expose `PROVIDER_SCOPES` which is list of `ProviderScope`
2. implement the `validate_scopes` function

**Review by shahargl — CHANGES_REQUESTED** [review]
hey @sc231997, nice work! Can you please add the scope validation?

You can look at jira_provider/grafana_provider for example.
You basically need two things:

Expose PROVIDER_SCOPES which is list of ProviderScope
implement the validate_scopes function
Also please add twillo-icon.png under keep-ui/public/icons

**Comment by sc231997** [comment]
> hey @sc231997, nice work! Can you please add the scope validation?
> 
> You can look at `jira_provider`/`grafana_provider` for example. You basically need two things:
> 
> 1. Expose `PROVIDER_SCOPES` which is list of `ProviderScope`
> 2. implement the `validate_scopes` function

Hi @shahargl 
Thanks!
I have a question related to `scopes`. Does `scope` here indicate weather the API token can send SMS or not? If yes, can you point me to Twilio docs, cannot find any related to that.

**Comment by shahargl** [comment]
@sc231997 basically we need any indication that the api token (or any other credentials) is ok.

I think the easiest way is to implement sending a message in validate_scopes.

So the scope should be something like "Send message" (message.send)
and validate scopes should be:
```python
...
try:
   # 15005550006 is a magic number according to https://www.twilio.com/docs/messaging/tutorials/automate-testing
   message = client.messages.create(body='All in the game, yo',  from_='+5571981265145', to='15005550006')
   scopes['message.read'] = True
except Exception as e:
    # validate the exception 
    if "authentication failed" in str(e)
        scopes['message.send'] = "Failed to send message - authentication failed"
    else:
      ...
      
```

File: keep/providers/twilio_provider/twilio_provider.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                        to="+5571981265131",
+                        body="scope test",
+                    )
+                    validated_scopes[scope.name] = True
+                except Exception as e:
+                    self.logger.warning(
+                        f"Failed to validate scope {scope.name}",
+                        extra={"reason": str(e)},
+                    )
+                    validated_scopes[scope.name] = False
```
Comment (shahargl) [inline keep/providers/twilio_provider/twilio_provider.py:83]:
```suggestion
                    validated_scopes[scope.name] = f"Failed to send a test SMS - {str(e)}"
```

File: keep/providers/twilio_provider/twilio_provider.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    def validate_scopes(self) -> dict[str, bool | str]:
+        validated_scopes = {}
+        for scope in self.PROVIDER_SCOPES:
+            if scope.name == "send_sms":
+                twilio_client = Client(
+                    self.authentication_config.account_sid,
+                    self.authentication_config.api_token,
+                )
+                try:
+                    # from: 15005550006 is a magic number according to https://www.twilio.com/docs/messaging/tutorials/automate-testing
```
Comment (shahargl) [inline keep/providers/twilio_provider/twilio_provider.py:71]: add debug log

File: keep/providers/twilio_provider/twilio_provider.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                    twilio_client.messages.create(
+                        from_="+15005550006",
+                        to="+5571981265131",
+                        body="scope test",
+                    )
+                    validated_scopes[scope.name] = True
```
Comment (shahargl) [inline keep/providers/twilio_provider/twilio_provider.py:77]: add debug log after ("test sms sent successfully")

**Review by shahargl — CHANGES_REQUESTED** [review]
minor changes

**Review by shahargl — APPROVED** [review]
lgtm

**Pull request merged**